bindings/generate_bindings.py

```diff
@@ -1,341 +1,313 @@
-                    raise AssertionError(f"bad type in query parameter {name}: {typ}")
-
-    def gen_function_param(self) -> Code:
-        if self.required:
-            typestr = self.type.annotation()
-            default = ""
-        else:
-            typestr = f'"typing.Optional[{self.type.annotation(prequoted=True)}]"'
-            default = " = None"
-        return f"    {self.name}: {typestr}{default},"
-
-    def gen_init_param(self) -> Code:
-        if self.required:
-            typestr = self.type.annotation()
-            default = ""
-        else:
-            typestr = f'"typing.Union[{self.type.annotation(prequoted=True)}, None, Unset]"'
-            default = " = _unset"
-        return f"    {self.name}: {typestr}{default},"
-
-    def dump(self, omit_unset: Code) -> Code:
-        return self.type.dump(self.name, omit_unset)
+def gen_init_param(param: swagger_types.Parameter) -> Code:
+    if param.required:
+        typestr = annotation(param.type)
+        default = ""
+    else:
+        typestr = f'"typing.Union[{annotation(param.type, prequoted=True)}, None, Unset]"'
+        default = " = _unset"
+    return f"    {param.name}: {typestr}{default},"
 
 
-class Class(TypeDef):
-    def __init__(self, name: str, params: typing.Dict[str, Parameter]):
-        self.name = name
-        self.params = params
+def gen_function_param(param: swagger_types.Parameter):
+    if param.required:
+        typestr = annotation(param.type)
+        default = ""
+    else:
+        typestr = f'"typing.Optional[{annotation(param.type, prequoted=True)}]"'
+        default = " = None"
+    return f"    {param.name}: {typestr}{default},"
 
-    def load(self, val: Code) -> Code:
-        return f"{self.name}.from_json({val})"
 
-    def dump(self, val: Code, omit_unset: Code) -> Code:
-        return f"{val}.to_json({omit_unset})"
-
-    def gen_def(self) -> Code:
-        required = sorted(p for p in self.params if self.params[p].required)
-        optional = sorted(p for p in self.params if not self.params[p].required)
-        out = [f"class {self.name}:"]
-        for k in optional:
-            v = self.params[k]
-            out += [f'    {k}: "typing.Optional[{v.type.annotation(prequoted=True)}]" = None']
+def is_streaming_response(defn: typing.Optional[swagger_types.TypeDef]):
+    if isinstance(defn, swagger_types.Class) and set(defn.params.keys()) == set(
+        ["result", "error"]
+    ):
+        error_ref = defn.params["error"]
+        return (
+            isinstance(error_ref.type, swagger_types.Ref)
+            and error_ref.type.name == "runtimeStreamError"
+        )
+    return False
+
+
+def unwrap_streaming_response(anno: swagger_types.TypeAnno):
+    if not isinstance(anno, swagger_types.Ref) or not is_streaming_response(anno.defn):
+        return anno
+    assert isinstance(anno.defn, swagger_types.Class)
+    return anno.defn.params["result"].type
+
+
+def gen_function(anno: swagger_types.Function) -> Code:
```

Comment:
prolly this parameter should be called `func: swagger_types.Function` rather than `anno`.  Or even just `f` since it's the central variable in the whole function.  But `anno` is a bit of a confusing name.